docs/build/wasm/ask_contracts.md

```diff
@@ -86,11 +225,21 @@ yarn && yarn build flipper.ts
 
 The above command will generate WASM code and the metadata file of the contract in `metadata.json` and `flipper.optimized.wasm` respectively.
 
-![08](img/08a.png)
+```
+ask-template
+├── asconfig.json               (assemblyscript config)
+├── askconfig.json              (ask-lang config)
+├── build
+│   └── metadata.json           generated by Ask!, configurable by `askconfig.json`
+│   └── flipper.optimized.wasm  generated by AssemblyScript, configurable by `asconfig.json`
+│   └── flipper.wat             generated by AssemblyScript, configurable by `asconfig.json`
+```
+
+## Deploy
 
 Now we will deploy this smart contract on our testnet.
 
-We will access [polkadot.js](https://polkadot.js.org/apps/) and deploy the smart contract. Select Shibuya testnet and pick `metadata.json` for “json for either ABI or .contract bundle” section and pick `flipper.opstimized.wasm` for “compiled contract WASM” section.
+We will access [polkadot.js](https://polkadot.js.org/apps/) and deploy the smart contract. Select Shibuya testnet and pick `metadata.json` for “json for either ABI or .contract bundle” section and pick `flipper.optimized.wasm` for “compiled contract WASM” section.
```

Comment:
I assume same issue will happen as with ink! Weight V2 issue?

like here: https://github.com/AstarNetwork/astar-docs/pull/256/files#diff-fe62424be3bde336c56fe184a4bfc9d1f87c027a64084fdffd8bc03ef11af5f7R24

Reply:
> I assume same issue will happen as with ink! Weight V2 issue?

Yes

- ask! does not produce `.contract` bundle, so it is not possible to deploy using [Contracts-UI](https://contracts-ui.substrate.io/) as it only supports contract bundle (I couldn't get it to accept wasm and metadata.json separately)
- other possibility would be deploy using polkadot.js and import contract in Contract-UI to interact, but maybe that would be too much for a complete newbie? thoughts?

Reply:
we combine wasm and metadata into .contract when we generate types for it with swanky.
Maybe we can add that to compile phase so that ask! also outputs .contract when compiled using swanky.